src/services/userToRoomJoiner.js

```diff
@@ -0,0 +1,61 @@
+const { User, Room } = require('../entities')
+
+class UserToRoomJoiner {
+    constructor({ userRepo, roomRepo }) {
+        this.userRepo = userRepo
+        this.roomRepo = roomRepo
+    }
+
+    call({ username, roomTitle }) {
+        try {
+            return this.join({ username, roomTitle })
+        } catch (error) {
+            return { error: error.message || error }
+        }
+    }
+
+    async join({ username, roomTitle }) {
+        const user = await this.getUser({ username })
+        const room = await this.getRoom({ title: roomTitle })
+        await this.addUserToRoom(room, user)
+        return { user, room }
+    }
+
+    async getUser({ username }) {
+        const user = new User({ username })
```

Comment:
Do I understand correctly that when passing `{ username }` it doesn't "exactly" fall into the User class, because the User class doesn't have anything to handle it.
But it goes further into the Base class, which has a constructor method?

Reply:
Exactly